bpfassets/libbpf/src/kepler.bpf.c

```diff
@@ -201,6 +201,18 @@ struct task_struct {
 SEC("tp_btf/sched_switch")
 int kepler_sched_switch_trace(u64 *ctx)
 {
+	// Skip some samples to minimize overhead
```

Comment:
We should not skip resetting the counters on a CPU during the schedule switch. Let’s consider an example to show why this is crucial:

Suppose task1 is running on CPU1, and we decide to skip the metric collection that resets the counters. Then, task2 starts running on CPU1, and we collect the counters at this point. The collected data will now include the hardware counters and CPU time for both task1 and task2. This will lead to inaccurate metrics, as the counters will reflect cumulative data for multiple tasks.

Therefore, we must reset the counters before skipping the schedule switch to ensure that each task's metrics are correctly isolated. Moving the counter reset to before collect_metrics_and_reset_counters would invalidate this approach, leading to inaccurate and misleading data.